ui/apps/platform/src/Containers/NetworkGraph/hooks/useScopeHierarchy.ts

```diff
@@ -1,24 +1,23 @@
-import useFetchClustersForPermissions from 'hooks/useFetchClustersForPermissions';
 import useURLSearch from 'hooks/useURLSearch';
+import { ClusterScopeObject } from 'services/RolesService';
 import { getScopeHierarchyFromSearch } from '../utils/hierarchyUtils';
 import { NetworkScopeHierarchy } from '../types/networkScopeHierarchy';
 
+const emptyScopeHierarchy = {
+    cluster: {
+        id: '',
+        name: '',
+    },
+    namespaces: [],
+    deployments: [],
+    remainingQuery: {},
+};
+
 /**
  * Returns the current scope hierarchy from the URL search params.
  */
-export function useScopeHierarchy(): NetworkScopeHierarchy {
+export function useScopeHierarchy(availableClusters: ClusterScopeObject[]): NetworkScopeHierarchy {
     const { searchFilter } = useURLSearch();
-    const { clusters } = useFetchClustersForPermissions(['NetworkGraph', 'Deployment']);
 
-    return (
-        getScopeHierarchyFromSearch(searchFilter, clusters) ?? {
-            cluster: {
-                id: '',
-                name: '',
-            },
-            namespaces: [],
-            deployments: [],
-            remainingQuery: {},
-        }
-    );
+    return getScopeHierarchyFromSearch(searchFilter, availableClusters) ?? emptyScopeHierarchy;
```

Comment:
Remembering the recent situation shared in a team meeting about accidental impure update to state, what do you think are pro and con:
* trust callers to clone and update scope hierarchy (if that even occurs in the code)
* call `cloneDeep(emptyScopeHierarchy)` here for return value

Reply:
I prefer the second: trust no one and we only have to clone once in the code. Although I don't think it is likely a caller will mutate the object, anything is possible.